README.md

```diff
@@ -1,123 +1,323 @@
+-Pargs=" --region=<region> \
+--project=${PROJECT_ID} \
+--streaming --enableStreamingEngine \
+--tempLocation=gs://${PROJECT_ID}-demo-data/temp \
+--numWorkers=1 --maxNumWorkers=2 \
+--runner=DataflowRunner \
+--filePattern=gs://${PROJECT_ID}-demo-data/*.csv \
+--dataset=${BQ_DATASET_NAME}   \
+--inspectTemplateName=${INSPECT_TEMPLATE_NAME} \
+--deidentifyTemplateName=${DEID_TEMPLATE_NAME} \
+--DLPMethod=DEID" 
+```
+You can run some quick [validations](https://cloud.google.com/solutions/validating-de-identified-data-bigquery-re-identifying-pii-data#validating_the_de-identified_dataset_in_bigquery) in BigQuery table to check on tokenized data.
+
+
+#### Re-Identification From BigQuery
+1. Export the Standard SQL Query to read data from bigQuery. One example from our solution guide:
 ```
+export QUERY="select ID,Card_Number,Card_Holders_Name from \`${PROJECT_ID}.${BQ_DATASET_NAME}.CCRecords_1564602828\` where safe_cast(Credit_Limit as int64)>100000 and safe_cast (Age as int64)>50 group by ID,Card_Number,Card_Holders_Name limit 10"
 ```
-gradle spotlessApply
+2. Create a gcs file with the query:
 
-gradle build
+```
+export REID_QUERY_BUCKET=<name>
+cat << EOF | gsutil cp - gs://${REID_QUERY_BUCKET}/reid_query.sql
+${QUERY}
+EOF
+```
+3. Create a pub-sub topic. Refer [create a topic](https://cloud.google.com/pubsub/docs/create-topic#create_a_topic)
+
+4. Run the pipeline by passing required parameters:
+```
+gradle run -DmainClass=com.google.swarm.tokenization.DLPTextToBigQueryStreamingV2 
+-Pargs="--region=<region> 
+--project=<project_id>
+--streaming --enableStreamingEngine 
+--tempLocation=gs://<bucket>/temp 
+--numWorkers=5 --maxNumWorkers=10 
+--runner=DataflowRunner 
+--tableRef=<project_id>:<dataset>.<table> 
+--dataset=<dataset> 
+--topic=projects/<project_id>/topics/<name> 
+--autoscalingAlgorithm=THROUGHPUT_BASED 
+--workerMachineType=n1-highmem-4 
+--deidentifyTemplateName=projects/<project_id>/deidentifyTemplates/<name> 
+--DLPMethod=REID 
+--keyRange=1024 
+--queryPath=gs://${REID_QUERY_BUCKET}/reid_query.sql"
 
-// inspect is default as DLP Method; For deid: --DLPMethod=DEID
-gradle run -DmainClass=com.google.swarm.tokenization.DLPTextToBigQueryStreamingV2 -Pargs="--region=<region> --project=<project_id> --streaming --enableStreamingEngine --tempLocation=gs://<bucket>/temp --numWorkers=1 --maxNumWorkers=2 --runner=DataflowRunner --filePattern=s3://<bucket>>/file.csv --dataset=<name>  --inspectTemplateName=<inspect_template> --deidentifyTemplateName=<deid_tmplate> --awsRegion=<aws_region> --awsCredentialsProvider=$AWS_CRED"
 ```
 
-## DeIdentification of JSONL files
-The pipeline supports JSONL file format where each line is a valid JSON Object and newline character is used to separate JSON objects. A sample file can be found in [test resources](src/test/resources/CCRecords_sample.jsonl).
+### Pipeline Parameters
+
+Following pipeline options have 
+
```

Comment:
Can we also call out that for the whole dataflow job these parameters will remain fixed once we start and autoscaling is enabled by default to process with maxNumber of workers as provided as maximum capacity?

Reply:
We don't think it is necessary to mention this here. Also the link to details about pipeline options is provided in the concepts section.

Reply:
Sounds good